### DISCUSSION TITLE
How to get keyless account from local storage

### DISCUSSION THREAD

**Discussion by ajaythxkur** [aptos-labs/aptos-developer-discussions#448] [General]
Api - https://api.testnet.aptoslabs.com/v1/tables/0x619dc29a0aac8fa146714058e8dd6d2d0f3bdf5f6331907bf91f3acd81e6935/item
Payload -
```
{
    "key": "0x1866655bc212e58829006595ca05ceeb4904eced2f6cbe7c4c64c030a67775b8",
    "key_type": "address",
    "value_type": "address"
}
```
Error -
```
{
    "message": "Table Item not found by Table handle(0x619dc29a0aac8fa146714058e8dd6d2d0f3bdf5f6331907bf91f3acd81e6935), Table key(\"0x1866655bc212e58829006595ca05ceeb4904eced2f6cbe7c4c64c030a67775b8\") and Ledger version(6018394684)",
    "error_code": "table_item_not_found",
    "vm_error_code": null
}
```

What is the fix to this?

**Comment by ajaythxkur** [comment]
This comes after redirectly from Google auth url

**Comment by magnum6actual** [comment]
I get this as well in browser console with every keyless login.  I haven't come across any issues - the client can still sign transactions.  Of course, I'd rather not have the error popping up in the console. It really grates on my OCD. But, presumably, the API is looking for something it can't find so there must be something not working as intended.

**Reply by ajaythxkur** [reply]
@magnum6actual does your wallet gets connected as mine does not.

**Reply by magnum6actual** [reply]
It does. I get the above error on:
```
     const derivedKeylessAccount = await aptos.deriveKeylessAccount({
        jwt: idToken,
        ephemeralKeyPair: ephemeralKeyPair,
      });
 ```
but I'm able to sign transactions with derivedKeylessAccount no problem.

**Reply by ajaythxkur** [reply]
I can see the account in localStorage. Is there any additional configuration to fetch it?

**Comment by ajaythxkur** [comment]


**Reply by ajaythxkur** [reply]
From integration guide
```
export const getLocalKeylessAccount = (): KeylessAccount | undefined => {
  try {
    const encodedAccount = localStorage.getItem("@aptos/account");
    return encodedAccount ? decodeKeylessAccount(encodedAccount) : undefined;
  } catch (error) {
    console.warn(
      "Failed to decode account from localStorage",
      error
    );
    return undefined;
  }
};
```

There are multiple accounts being stored in `@aptos-connect/keyless-accounts` in `accounts array` . Upon refreshing the page, i want my account to be connected, And on disconnecting localstorage to be empty so that it doesnt connect on next refresh

**Reply by magnum6actual** [reply]
What line of code is triggering the API 'table_item_not_found' error?

**Reply by ajaythxkur** [reply]
zustland automatically gets the account from localstorage? or we need to create an extra function for that?
